kernel/calls.c

```diff
@@ -272,6 +275,20 @@ void handle_interrupt(int interrupt) {
             .fault.addr = cpu->eip,
         };
         deliver_signal(current, SIGILL_, info);
+    } else if (interrupt == INT_BREAKPOINT) {
+        lock(&pids_lock);
+        send_signal(current, SIGTRAP_, (struct siginfo_) {
+            .sig = SIGTRAP_,
+            .code = SI_KERNEL_,
+        });
+        unlock(&pids_lock);
+    } else if (interrupt == INT_DEBUG) {
+        lock(&pids_lock);
```

Comment:
Shouldn't need to lock pids_lock, as the signal is being sent to current?

Reply:
`signal_delivery_stop` requires `pids_lock`: https://github.com/ish-app/ish/blob/f1eb124ee7559c83bdc584ac9c671b9203c39a27/kernel/signal.c#L109